hydromt_wflow/components/staticmaps.py

```diff
@@ -1,11 +1,254 @@
+        )
+
+    ## Mutating methods
+    def set(
+        self,
+        data: xr.DataArray | xr.Dataset,
+        name: str | None = None,
+        mask: str | xr.DataArray | None = None,
+    ):
+        """Add data to the staticmaps.
+
+        All layers of grid must have identical spatial coordinates.
+
+        Parameters
+        ----------
+        data : xr.DataArray | xr.Dataset
+            new map layer to add to grid
+        name : str, optional
+            Name of new map layer, this is used to overwrite the name of a DataArray
+            and ignored if data is a Dataset
+        mask : str | xr.DataArray, optional
+            A mask to clip
+        """
+        self._initialize_grid()
+        assert self._data is not None
+
+        name_required = isinstance(data, xr.DataArray) and data.name is None
+        if name is None and name_required:
+            raise ValueError(f"Unable to set {type(data).__name__} data without a name")
+        elif isinstance(data, xr.DataArray):
+            if name is not None:
+                data.name = name
+            data = data.to_dataset()
+        elif not isinstance(data, xr.Dataset):
+            raise ValueError(f"Cannot set data of type {type(data).__name__}")
+
+        # Check for cyclic data
+        if "time" in data.dims:
+            # Raise error if the dimension does not have a supported length
+            if len(data.time) not in [12, 365, 366]:
+                raise ValueError(
+                    f"Length of cyclic dataset ({len(data.time)}) is not supported by "
+                    "Wflow.jl. Ensure the data has length 12, 365, or 366"
+                )
+            tname = "time"
+            time_axes = {
+                k: v for k, v in self.data.sizes.items() if k.startswith("time")
+            }
+            if data["time"].size not in time_axes.values():
+                tname = f"time_{data['time'].size}" if "time" in time_axes else tname
+            else:
+                k = list(
+                    filter(lambda x: time_axes[x] == data["time"].size, time_axes)
+                )[0]
+                tname = k
+
+            if tname != "time":
+                data = data.rename_dims({"time": tname})
+        if "layer" in data.dims and "layer" in self.data:
+            if len(data["layer"]) != len(self.data["layer"]):
+                vars_to_drop = [
+                    var for var in self.data.data_vars if "layer" in self.data[var].dims
+                ]
+                # Drop variables
+                logger.warning(
+                    f"Replacing 'layer' coordinate, dropping variables \
+({vars_to_drop}) associated with old coordinate"
+                )
+                # Use `_grid` as `grid` cannot be set
+                self._data = self.data.drop_vars(vars_to_drop + ["layer"])
 
-__all__ = ["StaticmapsComponent"]
+        # Determine the masking layer
+        mask = get_mask_layer(mask, self.data, data)
```

Comment:
Is this really worth it to create this one use function with multiple args? In our case mask layer is quite fixed. I had imagined the function I had in utils could be re-used in processes or workflows as it was quite generic but for this one I wonder?

```suggestion
        mask_name = self.model._MAPS["basins"]
        if mask_name in self.data:
          mask = self.data[mask_name] != self.data[mask_name].raster.nodata
        elif mask_name in data:
          mask = data[mask_name] != data[mask_name].raster.nodata
        else:
          mask = None
```

Reply:
Ok. Then I would still for now maybe remove the mask argument and simplify to this:

```suggestion
        mask = get_mask_layer(self.model._MAPS["basins"], self.data, data)
```